shell/platform/android/io/flutter/plugin/platform/PlatformViewFactory.java

```diff
@@ -28,7 +28,7 @@ public PlatformViewFactory(@Nullable MessageCodec<Object> createArgsCodec) {
    *     null, or no arguments were sent from the Flutter app.
    */
   @NonNull
-  public abstract PlatformView create(Context context, int viewId, @Nullable Object args);
+  public abstract PlatformView create(Context context, long viewId, @Nullable Object args);
```

Comment:
This is a breaking change. Any advice?

Reply:
Hard-breaking every plugin that uses platform views is definitely a non-starter.

Is there some real problem that we need this change to solve? From what I see in the PR description and the issue, it's entirely a theoretical concern, rather than something that has actually caused problems. Presumably there are no Flutter apps that need more than several **billion** platform views over the life of the app, and I'm having a hard time imagining a scenario where some kind of non-sequential random mapping over more than the range of `int` would be required.

If this is all just theoretical, the easy solution would seem to be to throw if the value is too large, and document+assert on the Dart side that the value must be within the range of `int`.

Reply:
(E.g., it's also the case that on the Dart side `setSize` on a platform view takes a `Size`, whose width and height are `double`s, but I'm willing to bet that passing `double.maxFinite` for both would not end well. It's common for APIs to not accept every value that the types could allow for.)

Reply:
Yes, This is not a problem in practice, and given the breaking change, I am also more inclined to the solution you proposed.

Reply:
BTW, exposing the |viewId| to the user doesn't seem to make much sense.

Reply:
I didn't realized this was breaking. Stuart's suggestion seems like it is a good idea.

Reply:
> I didn't realized this was breaking. Stuart's suggestion seems like it is a good idea.

Sorry, that's my mistake. The breaking change is not obvious, and I forgot to mark it at the beginning.

Reply:
I have submitted a [new PR](https://github.com/flutter/flutter/pull/121203). Please review it when you have time. Thank you.

Reply:
السلام عليكم